hearth/lib/hearth/structure.rb

```diff
@@ -6,28 +6,34 @@ module Structure
     # Initializes the structure with the given options. If a member is
     # not provided, the default value is used.
     def initialize(options = {})
-      if respond_to?(:_defaults, true)
-        _defaults.each do |k, v|
-          options[k] = v unless options.include?(k)
+      if is_a?(Union)
+        super
+      else
+        if respond_to?(:_defaults, true)
+          _defaults.each do |k, v|
+            options[k] = v unless options.include?(k)
+          end
+        end
+        options.each do |k, v|
+          send("#{k}=", v)
```

Comment:
Is there a reason to not add a `:[]` and `:[]=` method to continue supporting the way we can access values in V3?

Reply:
Because union inherits struct and union shouldn't have those methods. I tried to keep it less hash like and more class like. I don't think we use the hash like interface anyway in v3. It should be a distinction between param and type.